frontend/src/features/codes/IndicesList.tsx

```diff
@@ -0,0 +1,258 @@
+                            setCurrentPage={setCurrentPage}
+                            pageInfo={data?.page}
+                        />
+                    </>
+                ) : (
+                    <LoadingSpinner />
+                )}
+            </QueryStateHandler>
+            <></>
+        </div>
+    );
+};
+
+const IndicesList = (): JSX.Element => {
+    const [currentIndexType, setCurrentIndexType] = useState(indexTypes[0]);
+    const [editMonth, setEditMonth] = useState<string | null>(null);
+    const [editValue, setEditValue] = useState<number | null>(null);
+    const initialSaveData: IIndex = {
+        indexType: indexTypes[0].label,
+        month: editMonth || "",
+        value: editValue || null,
+    };
+    const [formData, setFormData] = useImmer(initialSaveData);
+    const [editDialogOpen, setCreateDialogOpen] = useState<boolean>(false);
+    const initData = () => {
+        setEditMonth(null);
+        setEditValue(null);
+        setFormData(initialSaveData);
+    };
+    const closeDialog = () => {
+        initData();
+        setCreateDialogOpen(false);
+    };
+    const onSelectionChange = ({value}) => {
+        setEditMonth(null);
+        setEditValue(null);
+        setCurrentIndexType(() => ({label: value}));
+    };
+    return (
+        <div className="listing">
+            <div className="filters">
+                <Select
+                    label={"Indeksityyppi"}
+                    options={indexOptions}
+                    onChange={onSelectionChange}
+                    defaultValue={{
+                        value: indexTypes[0].label,
+                        label: getIndexTypeName(indexTypes[0].label),
+                    }}
+                />
+            </div>
+            <IndexResultList
+                indexType={currentIndexType}
+                setFormData={setFormData}
+                setCreateDialogOpen={setCreateDialogOpen}
+            />
+            <div className={"index-actions"}>
+                <Button
+                    theme="black"
+                    iconLeft={<IconPlus />}
+                    onClick={() => setCreateDialogOpen(true)}
+                >
+                    Lisää/päivitä indeksi
+                </Button>
+            </div>
+            <EditIndexDialog
+                editDialogOpen={editDialogOpen}
+                closeDialog={closeDialog}
+                indexType={currentIndexType}
+                formData={formData}
+                setFormData={setFormData}
+            />
+        </div>
+    );
+};
+
+const EditIndexDialog = ({indexType, formData, setFormData, editDialogOpen, closeDialog}) => {
+    const [saveIndex, {data: saveData, error: saveError, isLoading: isSaving}] = useSaveIndexMutation();
+    const handleSaveIndex = () => {
+        saveIndex({
+            data: formData,
+            index: indexType.label,
+            month: formData.month,
+        });
+        closeDialog();
+    };
```

Comment:
This doesn't handle error cases yet. If the user enters an invalid date the modal is closed and no error messages shown (Error message is shown on the modal but it's now hidden). Also no error toast notification is shown

Reply:
This should be fixed now. I made the dialog only close on a successful save - otherwise it should now stay open and show the errors next to the respective input field.